cylc/flow/pathutil.py

```diff
@@ -139,3 +139,66 @@ def make_suite_run_tree(suite):
         if dir_:
             os.makedirs(dir_, exist_ok=True)
             LOG.debug('%s: directory created', dir_)
+
+
+def make_localhost_symlinks(suite):
+    """Creates symlinks for any configured symlink dirs from glbl_cfg."""
+    dirs_to_symlink = get_dirs_to_symlink('localhost', suite)
+    rund = get_suite_run_dir(suite)
+    for key, value in dirs_to_symlink.items():
+        if key == 'run':
+            dst = rund
+        else:
+            dst = os.path.join(rund, key)
+        src = os.path.expandvars(value)
+        if '$' in src:
+            raise WorkflowFilesError(
+                f'Unable to create symlink to {src}. '
+                'Please check configuration.')
```

Comment:
Perhaps make it clear that this is due to environment variables, if this is caused by something daft like undefined `$TMPDIR` someone could end up scrabbling around for a while before they realise.